### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by sg3des** [tbogdala/fizzle#2]
Does your `fizzle` roadmap include an integration with physics engine(ex:your [cubez](https://github.com/tbogdala/cubez)), or you leave these separately?

Now it is a narrowly focused frameworks, it's not bad, but requires the creation of additional abstraction layer, for interaction of graphics and physics, and for many other things.

**Comment by tbogdala** [comment]
For now, I've been labelling fizzle as a "OpenGL rendering engine." This was a conscious choice. If I would have called it a game engine, that would imply integration of more libraries to give a more cohesive user experience.

My plan is to leave things separate. This give the most flexibility to the user who will be free to chose whether they want to use [cubez](http://www.github.com/tbogdala/cubez) or go with a library that re-implements [chipmunk](https://github.com/vova616/chipmunk) for 2d-only physics. I also have used engines before where I've thought that aspects of the engine were great but other aspects were essentially a showstopper because they didn't do what I needed and couldn't be substituted for something else.

The consequence is that you do have to glue things yourself. I figure that I'll never win the engine battle of convenience because many have come before me and have years of work put into them. I plan to offset this a bit by writing more samples and some tutorials.

The last part of this answer is that I'm still not sure I got some of the design where I want it yet. That's going to take me getting a bit farther in the [game I'm writing](http://gfycat.com/NauticalBowedBandicoot) to feel like what I'm doing is good enough. Most of the new features that have popped into fizzle have come from prototypes that I wrote and then decided to put the code into the core library.

This is a good question, and maybe I should get around to building a wiki page for this stuff.
